### PR TITLE
dev: Update minimum Node version requirement

### PR DESCRIPTION
**Pull request by nejraselimovic**
#### Summary
Updates the minimum required Node.js version in the DEVELOPMENT.md doc from 12.x to 14.x. 

#### Testing

I couldn't build the frontend assets until I updated the version according to @kschiffer instruction.

#### Checklist
- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [ ] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by kschiffer — APPROVED** [review]

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ The Things Stack components are primarily built in Go, we use React for web fron
 
 ## Development Environment
 
-The Things Network's development tooling uses [Mage](https://magefile.org/). Under the hood, `mage` calls other tools such as `git`, `go`, `yarn`, `docker` etc. Recent versions are supported; Node v12.x and Go v1.17x.
+The Things Network's development tooling uses [Mage](https://magefile.org/). Under the hood, `mage` calls other tools such as `git`, `go`, `yarn`, `docker` etc. Recent versions are supported; Node v14.x and Go v1.17x.
```
Comment (kschiffer) [inline DEVELOPMENT.md]:
```suggestion
The Things Network's development tooling uses [Mage](https://magefile.org/). Under the hood, `mage` calls other tools such as `git`, `go`, `yarn`, `docker` etc. Recent versions are supported; Node v14.x (minimum) and Go v1.17x.
```

**Review by htdvisser — COMMENTED** [review]
I think we should just recommend the exact same version as we use / our tooling uses, which is Node v16.x

**Comment by nejraselimovic** [comment]
So, I changed the Node version to 16.x as suggested. I don't understand why the PR is failing though, I performed a rebase on a newly pulled `v3.19` branch. Can someone help me resolve this?

**Review by htdvisser — APPROVED** [review]

**Comment by htdvisser** [comment]
The mergeability check is unrelated to this PR, so okay to ignore that.

**Pull request merged**